python/sglang/srt/managers/schedule_batch.py

```diff
@@ -452,9 +466,20 @@ def check_finished(self):
 
             for stop_str in self.sampling_params.stop_strs:
                 if stop_str in tail_str or stop_str in self.decoded_text:
-                    self.finished_reason = FINISH_MATCHED_STR(matched=stop_str)
+                    self.finished_reason = FINISH_MATCHED_REGEX(matched=stop_str)
                     return
 
+        # Check stop regex
+        if self.sampling_params.stop_regex_strs:
+            decode_res, new_text = self.get_next_inc_detokenization()
```

Comment:
It seems `get_next_inc_detokenization` will get partial detokenization results when e.g. jump forward happens, then it seems we will be checking regex using partial text instead of full text. A simplest fix may be to always detokenize the whole string.